doc/computing/parallelism.rst

```diff
@@ -317,11 +317,18 @@ Users looking for the best performance might want to tune this variable using
 powers of 2 so as to get the best parallelism behavior for their hardware,
 especially with respect to their caches' sizes.
 
-`SKLEARN_DOC_BUILD_WARNINGS_AS_ERRORS`
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-This environment variable issue errors instead of warnings when building the
-documentation. It ensures that we don't introduce new warnings in the example
-gallery. By default, the warnings are treated as errors (e.g. `"true"`). This
-is different from `SPHINXOPTS="-W"` that catch syntax warnings from the rst
-generation.
+`SKLEARN_WARNINGS_AS_ERRORS`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This environment variable is used to turn warnings into errors in two different places:
+
+- tests, for example by running `SKLEARN_WARNINGS_AS_ERRORS=1 pytest sklearn`.
+- documentation build, where it is turned on by default when you do `make
+  html`. You can do `SKLEARN_WARNINGS_AS_ERRORS=0 make html` if you want to
+  ignore warnings. Note that this checks that running examples don't produce
+  any warnings which is not the same as `SHPINXOPTS="-W"` that checks syntax
+  issues in the rst files.
```

Comment:
Based on the current diff it's set to 1 when building the doc but 0 when running the tests.

I think that for the sake of consistency, it should be 0 everywhere but we should set the `SKLEARN_WARNINGS_AS_ERRORS=1` explicitly when building the doc on the CI.

WDYT @adrinjalali @lesteve?

Reply:
+1 for having SKLEARN_WARNINGS_AS_ERRORS=0 in the doc build as well. I guess one reason to have it set to 1/true by default in the doc build was that people have the same settings as the CI by default.

As I mentioned above, my main reason to have `SKLEARN_WARNINGS_AS_ERRORS=0` is to avoid getting reports from contributors with warnings in some environments that are not worth fixing.

Reply:
I don't mind. And I can have the env present in my session always, so my personal workflow would be fine anyway.

Reply:
2 weeks ago, this error by default on local build was the source of confusion for an experience scikit-learn contributor because of a deprecation warning on Python 3.12 only.